blog-content/tracing-distributed-systems-in-nextjs.md

```diff
@@ -0,0 +1,214 @@
+
+Service-oriented architectures often create a web of HTTP calls, all of which should be connected back to the initial request.
+
+![Distributed tracing flame graph](/images/blog/tracing-distributed-systems-in-nextjs/trace-screenshot-with-callouts.png)
+
+## Approach
+
+The following example of Distributed Tracing is extracted from this repo:
+
+**[GitHub: Distributed Tracing Example](https://github.com/highlight/distributed-tracing-example)**
+
+We will demonstrate how to use Highlight to connect multiple services together by passing HTTP headers.
+
+- The web client initiates an HTTP `Fetch` request, passing the `x-highlight-request` header which contains a generated `sessionId` and `requestId`.
+- A Next.js API function receives that request, associates the `sessionId` and `requestId` to it's local trace, then passes the `x-highlight-request` header to a third service, this time written in Golang.
+- The Golang service receives the `x-highlight-request` header and associates it to the local trace.
+
+## Instrumenting the Next.js web client
+
+Client-side integration is as easy as injecting `<HighlightInit />` into your web app. 
+
+- Inject `<HighlightInit />` once, preferably in a `layout.tsx` file or `_app.tsx` file.
+- `<HighlightInit />` accepts props that correspond to the JavaScript SDK's [options arguments](https://www.highlight.io/docs/sdk/client#Hinit).
+- Explore Highlight's wide range of [configuration options](https://www.highlight.io/docs/getting-started/client-sdk/replay-configuration/overview), including 
+    - Canvas and WebGL support, 
+    - user identification,
+    - and privacy and redaction.
+
+```typescript
+// app/layout.tsx
+import { CONSTANTS } from "../constants";
+import { HighlightInit } from "@highlight-run/next/client";
+
+export default function RootLayout({
+  children,
+}: {
+  children: React.ReactNode;
+}) {
+  return (
+    <>
+      <HighlightInit
+        // excludedHostnames={['localhost']}
+        projectId={CONSTANTS.ENV.HIGHLIGHT.PROJECT_ID}
+        serviceName="my-nextjs-frontend"
+        tracingOrigins
+        networkRecording={{
+          enabled: true,
+          recordHeadersAndBody: true,
+        }}
+        debug
+      />
+
+      <html lang="en">
+        <body>{children}</body>
+      </html>
+    </>
+  );
+}
+```
+
+### Calling the Next.js API function
+
+Use the browser's `Fetch` API to call a Next.js API function. 
+
+Highlight monkey patches `window.fetch` and adds the `x-highlight-request` header automatically.
+
+See the full [list of monkey patches](https://www.highlight.io/docs/getting-started/client-sdk/replay-configuration/monkey-patches) in our docs.
+
+```jsx
+ <button
+    onClick={async () => {
+        await fetch("/propagation-test");       
+    }}
+>
+    Trigger propagation
+</button>
+```
```

Comment:
<img width="509" alt="Screenshot 2024-04-18 at 16 50 44" src="https://github.com/highlight/highlight/assets/1351531/d5a5234a-66e4-4ebe-a033-4033cde74c50">

this is doing something weird with the styling